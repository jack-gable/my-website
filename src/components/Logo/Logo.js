import React from "react";
import Link from "next/link";
import styles from "./Logo.module.css";
import { Pacifico } from "next/font/google";

const pacifico = Pacifico({ subsets: ["latin"], weight: ["400"] });

function Logo({ mobileAlignment = "left" }) {
	return (
		<Link
			className={styles.wrapper}
			href="/"
			data-mobile-alignment={mobileAlignment}
		>
			<div className={`${pacifico.className} ${styles.logo}`}>JG</div>
		</Link>
	);
}

export default Logo;
