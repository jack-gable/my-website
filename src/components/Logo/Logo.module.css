.wrapper {
	display: block;
	text-decoration: none;
}

.logo {
	font-size: 1.5rem;
	text-align: center;
	border: 2px solid;
	border-radius: 6px;
	padding: 1px 5px 5px 2px;
}

.logo:hover,
.logo:focus {
	background-color: var(--color-gray-0);
}
