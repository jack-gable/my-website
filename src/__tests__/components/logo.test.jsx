import React from "react";
import Logo from "@/components/Logo";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

describe("<Logo />", () => {
	it("renders the <Logo /> component with text", () => {
		const { container, getByText } = render(<Logo />);

		expect(getByText("JG")).toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});
});
