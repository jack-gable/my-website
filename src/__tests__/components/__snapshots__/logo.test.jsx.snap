// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> renders the <Logo /> component with text 1`] = `
<a
  class="wrapper"
  data-mobile-alignment="left"
  href="/"
>
  <div
    class="className logo"
  >
    JG
  </div>
</a>
`;
